### PR TITLE
Parallel operations of disk-ann file manager (#20120)

### PR DESCRIPTION
**Pull request by zhagnlu**
#20120

### CODE REVIEW

File: internal/core/src/storage/SafeQueue.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+        t = std::move(queue_.front());
+        queue_.pop();
+
+        return true;
+    }
+
+ private:
+    std::queue<T> queue_;
+    mutable std::mutex mutex_;
```
Comment (czs007) [inline internal/core/src/storage/SafeQueue.h:68]: use shard_mutex will be better.
[[share_mutex](https://en.cppreference.com/w/cpp/thread/shared_mutex)](https://en.cppreference.com/w/cpp/thread/shared_mutex)

**Comment by zhagnlu** [comment]
run-cpu-e2e

**Comment by zhagnlu** [comment]
rerun ut

**Pull request merged**